pkg/registries/registries.go

```diff
@@ -155,10 +187,32 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		reg.Blocked = true
 		for i := range config.Registries {
 			reg := &config.Registries[i]
-			if scopeMatchesRegistry(reg.Location, blockedReg) {
+			scope := reg.Location
+			// If Prefix is a valid wildcard entry, we set scope to Prefix
+			if isValidWildcard(reg.Prefix) {
+				scope = reg.Prefix
+			}
+			if scopeMatchesRegistry(scope, blockedReg) {
 				reg.Blocked = true
 			}
 		}
 	}
 	return nil
 }
+
+// isValidWildcard returns true if scope is a valid wildcard entry such as "*.registry.com"
+func isValidWildcard(scope string) bool {
```

Comment:
c/image does the following check for validation, which I think we should do here as well.
```golang
prefix[:2] == "*." && strings.ContainsAny(prefix, "/@:")
```

Reply:
Fixed